crates/experimentation_platform/src/api/experiments/handlers.rs

```diff
@@ -550,9 +552,31 @@ async fn list_experiments(
 
     let query_builder = |filters: &ExpListFilters| {
         let mut builder = experiments::experiments.into_boxed();
-        if let Some(states) = filters.status.clone() {
+        if let Some(ref states) = filters.status {
             builder = builder.filter(experiments::status.eq_any(states.0.clone()));
         }
+        if let Some(ref experiment_name) = filters.experiment_name {
+            builder =
+                builder.filter(experiments::name.like(format!("%{}%", experiment_name)));
+        }
+        if let Some(ref context_search) = filters.context {
+            builder = builder.filter(
+                sql::<Bool>("context::text LIKE ")
```

Comment:
what is this query doing here ?

Reply:
So the way we're storing contexts is like so:
```
{"and":[{"==":[{"var":"city"},"simple"]},{"==":[{"var":"vehicle_type"},"auto"]}]}
```

The query here is converting this (which is JSON in pg) to TEXT so we can use LIKE on it, for eg:
```
context::text LIKE '%{"==":[{"var":"city"},"simple"]}%'
```

Which would match any context that has city = simple as a dimension. It works for more than 1 dimension as well, where it does fail is if you specify `vehicle=auto and city=simple` but the experiment has `city=simple and vehicle=auto` (though `city=simple` or `vehicle=auto` would work). I want to see how folks use this feature, and if they encounter an issue I'll write the context search similar to how you've done it.